sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -207,6 +207,16 @@ class SessionCatalog(
           "you cannot create a database with this name.")
     }
     validateName(dbName)
+    // SPARK-25464 fail if DB location exists and is not empty
+    val dbPath = new Path(dbDefinition.locationUri)
+    val fs = dbPath.getFileSystem(hadoopConf)
+    if (fs.exists(dbPath)) {
+      val fileStatus = fs.listStatus(dbPath)
+      if (fileStatus.nonEmpty) {
+        throw new AnalysisException(
+          s"Cannot create Database to the location which is not empty.Given path is ${dbPath}")
```

Comment:
The message here has a few minor typos. Let's try: `s"Cannot create database at location $dbPath because it already exists."`

Reply:
if the folder exists and it is non-empty then we allow to create database,so this statement does not conflict the user ?

Reply:
That's fine, sure.